agents/locobot/locobot_agent.py

```diff
@@ -129,6 +134,64 @@ def objects_in_memory(sid):
         def log_interaction_data(sid, interactionData):
             self.interaction_logger.logInteraction(interactionData)
 
+        @sio.on("label_propagation")
+        def label_propagation(sid, postData): 
+                        
+            # Decode rgb map
+            rgb_bytes = base64.b64decode(postData["prevRgbImg"])
+            rgb_np = np.frombuffer(rgb_bytes, dtype=np.uint8)
+            rgb_bgr = cv2.imdecode(rgb_np, cv2.IMREAD_COLOR)
+            rgb = cv2.cvtColor(rgb_bgr, cv2.COLOR_BGR2RGB)
+            src_img = np.array(rgb)
+            height, width, _ = src_img.shape
+
+            # Convert depth map to meters
+            depth_imgs = []
+            for i, depth in enumerate([postData["prevDepth"], postData["depth"]]): 
+                depth_encoded = depth["depthImg"]
+                depth_bytes = base64.b64decode(depth_encoded)
+                depth_np = np.frombuffer(depth_bytes, dtype=np.uint8)
+                depth_decoded = cv2.imdecode(depth_np, cv2.IMREAD_COLOR)
+                depth_unscaled = (255 - np.copy(depth_decoded[:,:,0]))
+                depth_scaled = depth_unscaled / 255 * (float(depth["depthMax"]) - float(depth["depthMin"]))
+                depth_imgs.append(depth_scaled)
+            src_depth = np.array(depth_imgs[0])
+            cur_depth = np.array(depth_imgs[1])
+
+            # Convert mask points to mask maps then combine them
+            src_label = np.zeros((height, width)).astype(int)
+            for n, o in enumerate(postData["prevObjects"]): 
+                poly = Polygons(o["mask"])
+                bitmap = poly.mask(height, width)
+                # TODO probably a cleaner way to do this with numpy arrays
```

Comment:
you can replace this for loop with
`src_label[bitmap == True] = n+1`

Reply:
It turns out `bitmap` is not a np array and cannot be easily converted into one (afaik), so I'll leave this as is for now.